### PR TITLE
Revert k8s 1.17.9 upgrade

### PR DESCRIPTION
**Pull request by davidcassany**
## Why is this PR needed?

This is required in case we decide to backport the required cve in 4.2.2 instead of a patch level version bump of k8s.

## What does this PR do?

Reverts the k8s 1.17.9 upgrade, more specific it reverts the commits:
  * 5757a237c798d6b78d2b3e592b9aea2a14351bc4
  * 48f33d67d256e820951bc783fbb642ae2d5c1d34
  * 925f9381befa55d9e4fb9d8d8f00455e1e55c288

## Anything else a reviewer needs to know?

Special test cases, manual steps, links to resources or anything else that could be helpful to the reviewer.

## Info for QA

This is info for QA so that they can validate this. This is **mandatory** if this PR fixes a bug.
If this is a new feature, a good description in "What does this PR do" may be enough.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

How can we reproduce the issue? How can we see this issue? Please provide the steps and the prove
this issue is not fixed.

### Status **AFTER** applying the patch

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
but the documentation needs to be finalized before the PR can be merged. 

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,36 +9,36 @@ require (
 	golang.org/x/crypto v0.0.0-20200220183623-bac4c82f6975
 	golang.org/x/net v0.0.0-20191004110552-13f9640d40b9
 	golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45
-	k8s.io/api v0.17.9
-	k8s.io/apimachinery v0.17.9
-	k8s.io/client-go v0.17.9
-	k8s.io/cluster-bootstrap v0.17.9
+	k8s.io/api v0.17.5
```
Comment (chentex) [inline go.mod:12]: Shouldnl't it be 1.17.4? Or am I missing something?

Comment (davidcassany) [reply go.mod:12]: This is just the revert, on k8s 1.17.4 we already had this mismatch, I don't know why. I also noticed it when I upgraded it to 1.17.9

Comment (chentex) [reply go.mod:12]: wow! then all good

**Comment by davidcassany** [comment]
Note the tests for this are likely to fail even if we use the correct branches since the CI is not prepared to test downgrades.

**Pull request merged**